projects/transformers/trainer_mixins/one_cycle_lr.py

```diff
@@ -0,0 +1,85 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from torch.optim.lr_scheduler import OneCycleLR
+
+
+class OneCycleLRMixin:
+    """
+    Mixin to HF Trainer for using the `OneCycleLR`_. See documentation for argument
+    details.
+
+    .. _OneCycleLR:
+        https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.OneCycleLR
+    """
+
+    def __init__(
+        self,
+        max_lr=1e-2,
```

Comment:
We have to be careful that all these extra attributes are under trainer_kwargs, and they might start colliding. I prefixed the ones in KD with kd_. Not saying it is an ideal or prefered solution, just something to keep in mind.

Reply:
Fair point. We generally didn't do that in Vernon and we haven't run into issues yet. I'll keep the argument names as they are. But I'll keep it in mind moving forward.